File: 1664-ways-to-make-a-fair-array/1664-ways-to-make-a-fair-array.py
class Solution:
    def waysToMakeFair(self, nums: List[int]) -> int:
        
        l = len(nums)
        
        odd = list()
        even = list()
        
        osum = 0
        esum = 0
        
        for i in range(0,l):
            
            if i % 2 == 0:
                
                esum += nums[i]
            
            else:
                
                osum += nums[i]
                
            odd.append(osum)
            even.append(esum)
        
        ans = 0
        
        for i in range(0, l):
            
            if i == 0:
                nos = even[l-1] - nums[0]
                nes = odd[l-1]
                if nes == nos:
                    ans+=1
                continue
            
            nes = even[i-1] + odd[l-1] - odd[i]
            nos = odd[i-1] + even[l-1] - even[i]
            
            if nes == nos:
                ans+=1
                    
        return ans
        
        
        
        

File: 1664-ways-to-make-a-fair-array/1664-ways-to-make-a-fair-array.cpp
class Solution {
public:
    int waysToMakeFair(vector<int>& nums) {
        int n=nums.size();
        vector<int>oddSum(n,0);
        vector<int>evenSum(n,0);
        int even=0,odd=0;
        for(int i=0;i<n;i++){
            if(i%2==0){
                even+=nums[i];
            }
            else{
                odd+=nums[i];
            }
            evenSum[i]=even;
            oddSum[i]=odd;
        }
        int cnt=0;
        for(int i=0;i<n;i++){
            if(i==0){
                int nos=evenSum[n-1]-nums[0];
                int nes=oddSum[n-1];
                if(nos==nes){
                    cnt++;
                    
                }
                continue;  
            }
            int nes=evenSum[i-1]+oddSum[n-1]-oddSum[i];
            int nos=oddSum[i-1]+evenSum[n-1]-evenSum[i];            
            if(nes==nos){
                cnt++;
            }
        }
        return cnt;
    }
};